imbi/clients/gitlab.py

```diff
@@ -222,3 +236,34 @@ async def fetch_user_namespace(self):
             user_info['username'],
             response.code,
             title='GitLab API Failure')
+
+    async def _refresh_token(self) -> None:
+        body = urllib.parse.urlencode({
+            'grant_type': 'refresh_token',
+            'refresh_token': self.token.refresh_token,
+            'client_id': self.token.integration.client_id,
+            'client_secret': self.token.integration.client_secret,
+        })
+        self.logger.info('refreshing token for %s (%s)', self.user.username,
+                         self.token.external_id)
+        self.logger.debug('refresh_request: %r', body)
+        response = await super().http_fetch(
+            str(self.token.integration.token_endpoint),
+            method='POST',
+            body=body,
+            content_type='application/x-www-form-urlencoded',
+        )
```

Comment:
Can we do basic auth here? Similar to 
https://github.com/AWeber-Imbi/imbi-api/pull/68/commits/3ec775af97e5e35c58846be2d3bc5f8cd26ba844

Reply:
Lemme try that ... _they only supported IDs in request body in earlier versions_